libcudacxx/include/cuda/std/detail/libcxx/include/__cccl/ptx_isa.h

```diff
@@ -105,4 +105,13 @@
 #endif
 #endif // __cccl_ptx_isa >= 800
 
+// NVRTC uses its own <nv/target> header, so we need to manually tell it when we expect SM90a to be available
+#if defined(_CCCL_COMPILER_NVRTC) && !defined(NV_HAS_FEATURE_SM_90a)
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 900) && defined(__CUDA_ARCH_FEAT_SM90_ALL))
+#define NV_HAS_FEATURE_SM_90a NV_PROVIDES_SM_90
+#else // ^^^ SM90a ^^^ / vvv !SM90a vvv
+#define NV_HAS_FEATURE_SM_90a NV_NO_TARGET
+#endif //
+#endif // _CCCL_COMPILER_NVRTC && !NV_HAS_FEATURE_SM_90a
+
```

Comment:
Thanks! I love this hack. Now that we know that it works, should we perhaps move it to "our" nv/target in a separate PR?

Reply:
We cannot do that. The issue is that nvrtc does not scan *our* nv/target sources

Reply:
I mean: move this snippet from `ptx_isa.h` to our `nv/target` header?

Reply:
(finally) Makes sense.. Sorry for  being slow on the uptake, must be a lack of oxygen or something.. ^^